### PR TITLE
Update Inspection Script

### PR DESCRIPTION
**Pull request by giorgos-apo**
Stores memory, disk, distribution, uptime, vm information. Checks for… SELinux and docker installation and prints warnings. Stores kubernetes info about pv and pvc.

### CODE REVIEW

File: scripts/inspect.sh [hunk 1]
```diff
@@ -88,22 +132,74 @@ function suggest_fixes {
...
+  fi
+
+  # check for selinux. if enabled, print warning.
+  if getenforce | grep -q 'Enabled'
+  then
+    printf -- '\033[0;33m WARNING: \033[0m SElinux is enabled. Consider disabling it.\n'
+  fi
+
+  # check for docker
+  if [ -d "/etc/docker/" ]; then
```
Comment (ktsakalozos) [inline scripts/inspect.sh]: Could you test here that `localhost:32000` is not in `/etc/docker/`. You could do that with something like like  `if [ -d "/etc/docker/" ] && ! sudo grep localhost:32000 /etc/docker/daemon.json > /dev/null`

File: scripts/inspect.sh [hunk 2]
```diff
@@ -88,22 +132,74 @@ function suggest_fixes {
...
   if echo $ufw | grep "Status: active" &> /dev/null && ! echo $ufw | grep cbr0 &> /dev/null
   then
-      printf -- '\033[0;33m WARNING: \033[0m Firewall is enabled. Consider allowing pod traffic '
-      printf -- 'with: sudo ufw allow in on cbr0 && sudo ufw allow out on cbr0\n'
+    printf -- '\033[0;33m WARNING: \033[0m Firewall is enabled. Consider allowing pod traffic '
+    printf -- 'with: sudo ufw allow in on cbr0 && sudo ufw allow out on cbr0\n'
+  fi
+
+  # check for selinux. if enabled, print warning.
+  if getenforce | grep -q 'Enabled'
```
Comment (ktsakalozos) [inline scripts/inspect.sh]: Could we please do this: `if getenforce 2>&1 | grep  'Enabled' > /dev/null` so we do not print the stderr ?

File: scripts/inspect.sh [hunk 3]
```diff
@@ -88,22 +132,74 @@ function suggest_fixes {
...
+
+    # check for selinux. if enabled, print warning.
+    if getenforce | grep -q 'Enabled'
+    then
+    	printf -- '\033[0;33m WARNING: \033[0m SElinux is enabled. Consider disabling it.\n'
+    fi
+
+
+    # check for docker
+    if [ -d "/etc/docker/" ]; then
```
Comment (ktsakalozos) [inline scripts/inspect.sh]: This test seems to be a duplicate

File: scripts/inspect.sh [hunk 4]
```diff
@@ -88,22 +132,74 @@ function suggest_fixes {
...
+
+    ufw=$(ufw status)
+    if echo $ufw | grep "Status: active" &> /dev/null && ! echo $ufw | grep cbr0 &> /dev/null
+    then
+      printf -- '\033[0;33m WARNING: \033[0m Firewall is enabled. Consider allowing pod traffic '
+      printf -- 'with: sudo ufw allow in on cbr0 && sudo ufw allow out on cbr0\n'
+    fi
+
+    # check for selinux. if enabled, print warning.
+    if getenforce | grep -q 'Enabled'
```
Comment (ktsakalozos) [inline scripts/inspect.sh]: Duplicate test?

File: scripts/inspect.sh [hunk 5]
```diff
@@ -88,22 +132,74 @@ function suggest_fixes {
     if lsof -Pi :8080 -sTCP:LISTEN -t &> /dev/null
     then
       printf -- '\033[0;33m WARNING: \033[0m Port 8080 seems to be in use by another application.\n'
+        if lsof -Pi :8080 -sTCP:LISTEN -t &> /dev/null
+        then
+            printf -- '\033[0;33m WARNING: \033[0m Port 8080 seems to be in use by another application.\n'
+        fi
+    fi
+
+    if iptables -L | grep FORWARD | grep DROP &> /dev/null
```
Comment (ktsakalozos) [inline scripts/inspect.sh]: Probably something did not go well, this looks like a duplicate

**Comment by giorgos-apo** [comment]
@ktsakalozos I have done what you 've asked. I 've also solved the duplicate issues. Please have a look.

File: scripts/inspect.sh [hunk 6]
```diff
@@ -93,19 +137,34 @@ function suggest_fixes {
...
-}
 
+  # check for selinux. if enabled, print warning.
+  if getenforce 2>&1 | grep 'Enabled' > /dev/null
+  then
+    printf -- '\033[0;33m WARNING: \033[0m SElinux is enabled. Consider disabling it.\n'
+  fi
+
+  # check for docker
+  if [ -d "/etc/docker/" ] && ! sudo grep localhost:32000 /etc/docker/daemon.json > /dev/null; then
```
Comment (ktsakalozos) [inline scripts/inspect.sh]: This gave me the following ugly output:
```
grep: /etc/docker/daemon.json: No such file or directory
 WARNING:  Docker is installed. You should mark the registry as insecure. Add the following to /etc/docker/deamon.json : 
 {
         "insecure-registries" : ["localhost:32000"] 
 }
```
Please test for the existence of the file with `[ -f "/etc/docker/daemon.json" ]`

**Comment by ktsakalozos** [comment]
Instead of creating a separate directory for each info we gather could we put the  vm_name, disk_usage, uptime, lsb_release, openssl under sys. The reason for this is that these metrics have are system metrics with no "depth", so we might as well put them all under the sys category.

Thank you.

**Comment by giorgos-apo** [comment]
@ktsakalozos  You are right about the output issues. I 've fixed both errors. Please check.

**Comment by ktsakalozos** [comment]
LGTM +1 Thank you @giorgos-apo

**Pull request merged**